sdk/servicebus/service-bus/test/batchReceiver.spec.ts

```diff
@@ -1135,3 +1136,34 @@ describe("Batch Receiver - Others", function(): void {
     await testAskForMore(true);
   });
 });
+
+describe("Batch Receiver - drain", function(): void {
+  afterEach(async () => {
+    await afterEachTest();
+  });
+
+  it("honors timeout", async () => {
+    await beforeEachTest(TestClientType.UnpartitionedQueue, TestClientType.UnpartitionedQueue);
+
+    // initialize receiver link to make it easier to test
+    await receiver.receiveMessages(1, 1);
+
+    if (!(receiver as any)._context.batchingReceiver.isOpen()) {
+      throw new Error(`Unable to initialize receiver link.`);
+    }
+
+    // Since the receiver has already been initialized,
+    // the `receiver_drained` handler is attached as soon
+    // as receiveMessages is invoked.
+    setTimeout(() => {
+      // remove `receiver_drained` event
+      (receiver as any)._context.batchingReceiver._receiver.removeAllListeners(
+        ReceiverEvents.receiverDrained
+      );
+    }, 0);
+
+    const results = await receiver.receiveMessages(1, 1);
```

Comment:
I think the comment could use a little more to it - I had to read this a few times before I got it (and even then...did I truly get it?)

If I did here's what I think you're testing:

1. The fix was to make sure that if receiver_drained was never called that we'd at least time out and still return the messages to the user
2. To simulate that you've removed the event listener, resulting in the listener not being called.
3. If messages _still _return then we assume it's because the delay() call fired off and caused us to return.

Is that correct?

Reply:
(btw, this feels like something that would have been better suited to a unit test. It's not easy in the current code-base to get those in but it's worth looking at how we might structure things better in track 2 to support it).

What do you think?

Reply:
Sure, I can add more context to the comment. You've got the gist of it. 

Also to make sure that we can actually remove the `receiver_drain` event handler, I had to make sure the link was already initialized because otherwise I'd be removing the listener before it was actually attached...rendering my test useless.

And yes, I do think this would be best as a unit test. Given the state of the v1 code-base, this was the best I could come up with without a lot more mocking of private fields in place.

Reply:
Yeah, I'm totally there with you - there's only so much you can change reasonably.